agent/bench-scripts/pbench-trafficgen

```diff
@@ -1063,214 +1152,191 @@ function inner_most_loop() {
-
-							# Move the tool data collected for the tool-group to its final resting place.
-							mv ${tt_benchmark_results_dir}/tools-${tool_group} ${benchmark_results_dir}/
-							if [[ ${?} -ne 0 ]]; then
-								error_log "INTERNAL ERROR: failed to move tool-trigger data, '${tt_benchmark_results_dir}/tools-${tool_group}' to '${benchmark_results_dir}'"
-								exit 1
-							fi
-							# The "tool-trigger" sample directory should be empty now.
-							rmdir ${tt_benchmark_results_dir}
-							if [[ ${?} -ne 0 ]]; then
-								error_log "INTERNAL ERROR: failed to remove tool-trigger results directory, '${tt_benchmark_results_dir}'"
-								exit 1
-							fi
-						else
-							msg="tool triggers did not fire for iteration/sample, '${iteration}/sample${sample}'"
-							echo "${msg}"; log "${msg}"
+					if [[ -e "${tt_benchmark_results_dir}/tools-${tool_group}" ]]; then
+						# Post-process the collected tool data now
+						pbench-postprocess-tools --group=${tool_group} --dir=${tt_benchmark_results_dir}
+
+						# Move the tool data collected for the tool-group to its final resting place.
+						mv ${tt_benchmark_results_dir}/tools-${tool_group} ${benchmark_results_dir}/
+						if [[ ${?} -ne 0 ]]; then
+							error_log "INTERNAL ERROR: failed to move tool-trigger data, '${tt_benchmark_results_dir}/tools-${tool_group}' to '${benchmark_results_dir}'"
+							exit 1
 						fi
-					else
-						if [[ ! -d $benchmark_results_dir ]]; then
-							error_log "Results directory $benchmark_results_dir does not exist, skipping post-processing"
-							continue
+						# The "tool-trigger" sample directory should be empty now.
+						rmdir ${tt_benchmark_results_dir}
+						if [[ ${?} -ne 0 ]]; then
+							error_log "INTERNAL ERROR: failed to remove tool-trigger results directory, '${tt_benchmark_results_dir}'"
+							exit 1
 						fi
-						echo "Not going to run $benchmark_name.  Only postprocesing existing data"
-						log "Not going to run $benchmark_name.  Only postprocesing existing data"
+					else
+						echo_log "tool triggers did not fire for iteration/sample, '${iteration}/sample${sample}'"
 					fi
-					if [ $rc -eq 0 ]; then
-						echo "$script_path/postprocess/$benchmark_name-postprocess \"$benchmark_results_dir\" \"$iteration\" \"$max_loss_pct\" \"$trafficgen_version\" \"$tool_label_pattern\" \"$tool_group\"" >"$benchmark_results_dir/$benchmark_name-postprocess.cmd"
-						chmod +x "$benchmark_results_dir/$benchmark_name-postprocess.cmd"
-						$benchmark_results_dir/$benchmark_name-postprocess.cmd 2>&1 | tee $benchmark_results_dir/$benchmark_name-postprocess.out || rc=1
+				else
+					if [[ ! -d $benchmark_results_dir ]]; then
+						error_log "Results directory $benchmark_results_dir does not exist, skipping post-processing"
+						continue
 					fi
-				done
+					echo_log "Not going to run $benchmark_name.  Only postprocesing existing data"
+				fi
```

Comment:
Lines 1227 - 1267 are the `if / else / fi` for post-processing-only or not.  At line 1228 we create the `${benchmark_results_dir}` (which arguably should not be `mkdir -p`, but `mkdir` with a check for failure).  At line 1262 we just check for a missing `${benchmark_results_dir}` so that we can skip the benchmark post-processing step.  At line 1266 we only want to indicate that we didn't run the benchmark, but will post-process the benchmark data below at line 1271.

Reply:
I must have missed the `continue` statement when I made this comment.  (This code is, sadly, not easy to deal with.)